controller/workitem.go

```diff
@@ -465,21 +466,26 @@ func ConvertJSONAPIToWorkItem(appl application.Application, source app.WorkItem,
 			target.Fields[workitem.SystemIteration] = iterationUUID.String()
 		}
 	}
-	if source.Relationships != nil && source.Relationships.Area != nil {
-		if source.Relationships.Area.Data == nil {
-			delete(target.Fields, workitem.SystemArea)
-		} else {
-			d := source.Relationships.Area.Data
-			areaUUID, err := uuid.FromString(*d.ID)
-			if err != nil {
-				return errors.NewBadParameterError("data.relationships.area.data.id", *d.ID)
-			}
-			if _, err = appl.Areas().Load(context.Background(), areaUUID); err != nil {
-				return errors.NewBadParameterError("data.relationships.area.data.id", *d.ID)
-			}
-			target.Fields[workitem.SystemArea] = areaUUID.String()
+
+	if source.Relationships != nil && source.Relationships.Area != nil && source.Relationships.Area.Data != nil {
+		d := source.Relationships.Area.Data
+		areaUUID, err := uuid.FromString(*d.ID)
+		if err != nil {
+			return errors.NewBadParameterError("data.relationships.area.data.id", *d.ID)
+		}
+		if _, err = appl.Areas().Load(context.Background(), areaUUID); err != nil {
+			return errors.NewBadParameterError("data.relationships.area.data.id", *d.ID)
 		}
+		target.Fields[workitem.SystemArea] = areaUUID.String()
+	} else if appl != nil && (source.Relationships == nil || source.Relationships.Area == nil || source.Relationships.Area.Data == nil) {
```

Comment:
This condition should be 
`source.Relationships != nil && source.Relationships.Area != nil && source.Relationships.Area.Data == nil`
because if `source.Relationships == nil ` we shouldn't be worrying about area at all.

In fact, a better way could be to use the previous structure:

```
if source.Relationships != nil && source.Relationships.Area != nil {
		if source.Relationships.Area.Data == nil {
			//delete(target.Fields, workitem.SystemArea)
                        /** 
                                your new code goes here.
                        */
		} else {
			/*
                                Existing code ..
                        */
		}
	}
```